packages/patternfly-3/react-console/src/ConsoleSelector/ConsoleSelector.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Grid, Row, Col, Form, FormGroup, Dropdown, MenuItem } from 'patternfly-react';
+import { Checkbox } from 'react-bootstrap';
+
+import { NONE_TYPE, SERIALCONSOLE_TYPE, VNCCONSOLE_TYPE } from './constants';
+
+class ConsoleSelector extends React.Component {
+  state = {
+    type: NONE_TYPE,
+    disconnectByChange: this.props.disconnectByChange,
+    keptConnection: {} // no connection exists when mounted
+  };
+
+  onTypeChange(type) {
+    this.setState(prevState => {
+      const keptConnection = prevState.disconnectByChange
+        ? { [type]: true }
+        : Object.assign({}, prevState.keptConnection, { [type]: true });
+
+      return {
+        type,
+        keptConnection
+      };
+    });
+  }
+
+  onChangeDisconnectBySwitchClick(target) {
+    this.setState(prevState => {
+      const actualType = prevState.keptConnection[prevState.type];
+      return {
+        disconnectByChange: target.checked,
+        keptConnection: target.checked ? { [actualType]: true } : prevState.keptConnection
+      };
+    });
+  }
+
+  getSelectedConsole() {
+    return this.getConsoleForType(this.state.type);
+  }
+
+  getConsoleForType(type) {
+    if (!this.props.children) {
+      return null;
+    }
+
+    const getChildTypeName = child => (child.props.type ? child.props.type : (child.type && child.type.name) || null);
+    const isChildOfType = child => getChildTypeName(child) === type;
+
+    // To keep connection, render all consoles but hide those unused
+    return React.Children.map(
+      this.props.children,
+      child =>
+        this.state.keptConnection[getChildTypeName(child)] ? (
```

Comment:
I prefer something like
```js
child && (<div...>)
```
this way you can avoid hiding things vs even putting them on the virtual dom?

Reply:
The point here is, I need to render and hide a child - to avoid unmounting it, so connection is kept.

So, all children are iterated through and those "already connected" are rendered with "hidden" attribute, so React will not unmount them.

Farther, the ternary operator is used to return "null" instead of "false". I don't prefer chaining '&&' with yet another '||' to do so due to readability.